ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerUtils.java

```diff
@@ -34,47 +37,39 @@
   private ServerUtils() {
   }
 
-  public static void handleError(final HttpServerResponse response, final int statusCode,
-      final int errorCode, final String errMsg) {
-    final ErrorResponse errorResponse = new ErrorResponse(errorCode, errMsg);
-    final Buffer buffer = errorResponse.toBuffer();
-    response.setStatusCode(statusCode).end(buffer);
```

Comment:
Before this refactor, `handleError()` was populating the status code and providing a message when ending the response, but the new code that previously called this method does not. Was this a conscious decision? I think it's nice to populate the response because otherwise failures appear silent from curl (and probably other tools as well?).

Reply:
Not sure I follow.. could you elaborate a bit?

Reply:
The old code (the line I've commented on here) did:
```
response.setStatusCode(statusCode).end(buffer);
```
whereas the new code does:
```
routingContext.fail(statusCode, exception);
```
The latter doesn't populate the response status code and message (buffer), right? As a result, failed curl requests come up blank. I remember being really confused by this when doing manual testing.

Reply:
Could you point me to the routingContext.fail() line you are referring to? There are probably a few places in the code that call that :)

RoutingContext fail will cause the failureHandler to be called.

Reply:
Ah I see, and both the regular failure handler (in `ServerVerticle`) and the old API failure handler (in `PortedEndpoints`) set the status code and message, so we should be good. Thanks for clarifying!

Reply:
Right, the change here basically centralises all the failure handling in one place so we don't end responses with failure from multiple places any more, which makes it a bit simpler to manage.